utils/gitlab_api.py

```diff
@@ -102,6 +116,41 @@ def create_delete_user_mr(self, username, paths):
 
         self.create_mr(branch_name, target_branch, title, labels=LABELS)
 
+    def create_delete_aws_access_key_mr(self, account, path, key):
+        prefix = 'qontract-reconcile'
+        target_branch = 'master'
+        branch_name = '{}-delete-aws-access-key-{}-{}-{}'.format(
+            prefix,
+            account,
+            key,
+            str(uuid.uuid4())[0:6]
+        )
+        title = '[{}] delete {} access key {}'.format(prefix, account, key)
+
+        if self.mr_exists(title):
+            return
+
+        self.create_branch(branch_name, target_branch)
+
+        f = self.project.files.get(file_path=path, ref=target_branch)
+        content = yaml.load(f.decode(), Loader=yaml.RoundTripLoader)
+        content.setdefault('deleteKeys', [])
+        content['deleteKeys'].append(key)
+        new_content = '---\n' + \
+            yaml.dump(content, Dumper=yaml.RoundTripDumper)
+        try:
+            self.update_file(branch_name, path, title, new_content)
+        except gitlab.exceptions.GitlabCreateError as e:
+            self.delete_branch(branch_name)
+            if str(e) != "400: A file with this name doesn't exist":
+                raise e
+            logging.info(
+                "File {} does not exist, not opening MR".format(path)
+            )
+            return
+
+        self.create_mr(branch_name, target_branch, title)
```

Comment:
no exception handling here?

Reply:
not at the moment. we have never seen this one fail before, so i'd like to just let a potential exception propogate at this point.

we can add exception handling to `create_mr` when we meet one. wdyt?

Reply:
sgtm